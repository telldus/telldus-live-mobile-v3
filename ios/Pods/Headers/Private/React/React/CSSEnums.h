../../../../../../node_modules/react-native/React/CSSLayout/CSSEnums.h