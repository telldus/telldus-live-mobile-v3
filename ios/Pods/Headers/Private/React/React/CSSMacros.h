../../../../../../node_modules/react-native/React/CSSLayout/CSSMacros.h