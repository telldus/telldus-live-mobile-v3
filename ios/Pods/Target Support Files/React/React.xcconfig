CLANG_CXX_LANGUAGE_STANDARD = c++14
CONFIGURATION_BUILD_DIR = $PODS_CONFIGURATION_BUILD_DIR/React
GCC_PREPROCESSOR_DEFINITIONS = $(inherited) COCOAPODS=1
HEADER_SEARCH_PATHS = "${PODS_ROOT}/Headers/Private" "${PODS_ROOT}/Headers/Private/React" "${PODS_ROOT}/Headers/Public" "${PODS_ROOT}/Headers/Public/React"
OTHER_LDFLAGS = -l"stdc++" -framework "JavaScriptCore"
PODS_BUILD_DIR = $BUILD_DIR
PODS_CONFIGURATION_BUILD_DIR = $PODS_BUILD_DIR/$(CONFIGURATION)$(EFFECTIVE_PLATFORM_NAME)
PODS_ROOT = ${SRCROOT}
PRODUCT_BUNDLE_IDENTIFIER = org.cocoapods.${PRODUCT_NAME:rfc1034identifier}
SKIP_INSTALL = YES
