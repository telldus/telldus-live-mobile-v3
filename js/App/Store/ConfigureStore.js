/**
 * Copyright 2016-present Telldus Technologies AB.
 *
 * This file is part of the Telldus Live! app.
 *
 * Telldus Live! app is free : you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Telldus Live! app is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Telldus Live! app.  If not, see <http://www.gnu.org/licenses/>.
 */

'use strict';

import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import promise from './Promise';
import array from './Array';
import analytics from './Analytics';
import reducers from 'Reducers';
import createLogger from 'redux-logger';
import { persistStore, autoRehydrate } from 'redux-persist';
import { AsyncStorage } from 'react-native';

let isDebuggingInChrome = __DEV__ && !!window.navigator.userAgent;

let logger = createLogger({
	predicate: (getState, action) => isDebuggingInChrome,
	collapsed: true,
	duration: true,
});

let createTheStore = applyMiddleware(thunk, promise, array, analytics, logger)(createStore);

let _store;
export function configureStore(onComplete: ?() => void) {
	const store = autoRehydrate()(createTheStore)(reducers);
	persistStore(store, {storage: AsyncStorage}, onComplete);
	if (isDebuggingInChrome) {
		window.store = store;
	}
	_store = store; // TODO: fix this ugly stuff
	return store;
}
export function getStore() { return _store; }
