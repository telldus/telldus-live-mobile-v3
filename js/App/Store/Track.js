/**
 * Copyright 2016-present Telldus Technologies AB.
 *
 * This file is part of the Telldus Live! app.
 *
 * Telldus Live! app is free : you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Telldus Live! app is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Telldus Live! app.  If not, see <http://www.gnu.org/licenses/>.
 */

'use strict';

import type { Action } from '../actions/types';
import DeviceInfo from 'react-native-device-info';

import { Analytics, Hits as GAHits } from 'react-native-google-analytics';
import { googleAnalyticsId } from 'Config';

function track(action: Action): void {

	let clientId = DeviceInfo.getUniqueID();
	let ga = new Analytics(googleAnalyticsId, clientId, 1, DeviceInfo.getUserAgent());

	if (action.type === 'SWITCH_TAB') {
		let screenView = new GAHits.ScreenView(
			'Telldus Live! app',
			action.tab,
			DeviceInfo.getVersion(),
			DeviceInfo.getBundleId()
		);
		ga.send(screenView);
	} else {
		let gaEvent = new GAHits.Event('Action', action.type);
		ga.send(gaEvent);
	}
}

module.exports = track;
