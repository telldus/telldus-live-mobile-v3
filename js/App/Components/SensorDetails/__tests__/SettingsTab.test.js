/**
 * Copyright 2016-present Telldus Technologies AB.
 *
 * This file is part of the Telldus Live! app.
 *
 * Telldus Live! app is free : you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Telldus Live! app is distributed in the hope this it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Telldus Live! app.  If not, see <http://www.gnu.org/licenses/>.
 *
 */


import React from 'react';
import {
	Dimensions,
} from 'react-native';
import {act} from 'react-test-renderer';

import { configureStore } from '../../../Store/ConfigureStore';
import {
	rendererWithIntlAndReduxProviders,
	DUMMY_SENSOR,
	withScreenPropsHOC,
} from '../../../../Utils/jestUtils';

import SettingsTab from '../SettingsTab';
import {
	setAppLayout,
	onReceivedSensors,
} from '../../../Actions';

let {height, width} = Dimensions.get('window');

const store = configureStore().store;

jest.useFakeTimers();


const onOp = () => {
};

describe('<SettingsTab /> - snapshot', () => {
	it('renders Sensor SettingsTab', () => {
		let component;
		act(() => {
			store.dispatch(setAppLayout({
				height,
				width,
			}));

			const sensors = {
				sensor: [
					DUMMY_SENSOR,
				],
			};
			store.dispatch(onReceivedSensors(sensors));

			const WithIntl = withScreenPropsHOC(
				<SettingsTab
					dispatch={onOp}
					route={{params: {
						id: DUMMY_SENSOR.id,
					}}}
					navigation={{}}
					currentScreen={'SSettings'}
					onAddToDashboard={onOp}
					onRemoveFromDashboard={onOp}
					sendSocketMessage={onOp}/>
			);

			component = rendererWithIntlAndReduxProviders(
				<WithIntl/>
			);
		});

		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
