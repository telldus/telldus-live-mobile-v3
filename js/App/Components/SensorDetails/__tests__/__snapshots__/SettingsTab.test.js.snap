// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SettingsTab /> - snapshot renders Sensor SettingsTab 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "flexGrow": 1,
    }
  }
  style={
    Object {
      "backgroundColor": "#eeeeee",
      "flex": 1,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
          "paddingBottom": 20.83275,
          "paddingHorizontal": 20.83275,
          "paddingTop": 10.416375,
        }
      }
    >
      <View
        accessibilityLabel="Name, Aeon Multisensor 6 , Double tap to edit"
        accessible={true}
        importantForAccessibility="yes"
        style={
          Array [
            Object {
              "backgroundColor": "#f5f5f5",
              "borderRadius": 2,
              "elevation": 2,
              "marginTop": 10.416375,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 1,
                "width": 0,
              },
              "shadowOpacity": 0.23,
              "shadowRadius": 2,
            },
            undefined,
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onLayout={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          pointerEvents="box-only"
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderRadius": 2,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "overflow": "hidden",
              "padding": 30,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "flexDirection": "row",
                "justifyContent": "flex-start",
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#000",
                    "fontFamily": "SuisseIntl",
                    "fontSize": 30,
                    "justifyContent": "center",
                  },
                  undefined,
                ]
              }
            >
              Name
            </Text>
          </View>
          <View
            accessible={true}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "flex-end",
                "opacity": 1,
                "paddingVertical": 8,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#4C4C4C",
                    "flex": 1,
                    "fontFamily": "SuisseIntl",
                    "fontSize": 30,
                    "textAlign": "right",
                    "textAlignVertical": "center",
                  },
                  undefined,
                ]
              }
            >
              Aeon Multisensor 6
               
            </Text>
          </View>
          <Image
            source={
              Object {
                "uri": "right_arrow_key",
              }
            }
            style={
              Object {
                "height": 30,
                "marginLeft": 30,
                "tintColor": "#BDBDBD",
                "width": 30,
              }
            }
          />
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "bottom": 0,
                  "left": 0,
                  "overflow": "hidden",
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                Object {
                  "borderRadius": 0,
                },
              ]
            }
          />
        </View>
      </View>
      <View
        accessibilityLabel="Show on dashboard,  , "
        accessible={false}
        importantForAccessibility="yes"
        style={
          Array [
            Object {
              "backgroundColor": "#f5f5f5",
              "borderRadius": 2,
              "elevation": 2,
              "marginTop": 10.416375,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 1,
                "width": 0,
              },
              "shadowOpacity": 0.23,
              "shadowRadius": 2,
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderRadius": 2,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "overflow": "hidden",
                "padding": 30,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "flexDirection": "row",
                "justifyContent": "flex-start",
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#000",
                    "fontFamily": "SuisseIntl",
                    "fontSize": 30,
                    "justifyContent": "center",
                  },
                  undefined,
                ]
              }
            >
              Show on dashboard
            </Text>
          </View>
          <RCTSwitch
            accessibilityRole="switch"
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onTintColor="#ccc"
            style={
              Array [
                Object {
                  "height": 31,
                  "width": 51,
                },
                Array [
                  Array [
                    Object {
                      "justifyContent": "flex-end",
                    },
                    undefined,
                  ],
                  Object {
                    "backgroundColor": "#cccccc",
                    "borderRadius": 16,
                  },
                ],
              ]
            }
            thumbTintColor="#999999"
            tintColor="#cccccc"
            value={false}
          />
        </View>
      </View>
      <View
        accessibilityLabel="Hide from sensor list,  , "
        accessible={false}
        importantForAccessibility="yes"
        style={
          Array [
            Object {
              "backgroundColor": "#f5f5f5",
              "borderRadius": 2,
              "elevation": 2,
              "marginTop": 10.416375,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 1,
                "width": 0,
              },
              "shadowOpacity": 0.23,
              "shadowRadius": 2,
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderRadius": 2,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "overflow": "hidden",
                "padding": 30,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "flexDirection": "row",
                "justifyContent": "flex-start",
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#000",
                    "fontFamily": "SuisseIntl",
                    "fontSize": 30,
                    "justifyContent": "center",
                  },
                  undefined,
                ]
              }
            >
              Hide from sensor list
            </Text>
          </View>
          <RCTSwitch
            accessibilityRole="switch"
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onTintColor="#ccc"
            style={
              Array [
                Object {
                  "height": 31,
                  "width": 51,
                },
                Array [
                  Array [
                    Object {
                      "justifyContent": "flex-end",
                    },
                    undefined,
                  ],
                  Object {
                    "backgroundColor": "#cccccc",
                    "borderRadius": 16,
                  },
                ],
              ]
            }
            thumbTintColor="#999999"
            tintColor="#cccccc"
            value={false}
          />
        </View>
      </View>
      <View
        accessibilityLabel="Store history,  , "
        accessible={false}
        importantForAccessibility="yes"
        style={
          Array [
            Object {
              "backgroundColor": "#f5f5f5",
              "borderRadius": 2,
              "elevation": 2,
              "marginTop": 10.416375,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 1,
                "width": 0,
              },
              "shadowOpacity": 0.23,
              "shadowRadius": 2,
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderRadius": 2,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "overflow": "hidden",
                "padding": 30,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "flexDirection": "row",
                "justifyContent": "flex-start",
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#000",
                    "fontFamily": "SuisseIntl",
                    "fontSize": 30,
                    "justifyContent": "center",
                  },
                  undefined,
                ]
              }
            >
              Store history
            </Text>
          </View>
          <RCTSwitch
            accessibilityRole="switch"
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onTintColor="#ccc"
            style={
              Array [
                Object {
                  "height": 31,
                  "width": 51,
                },
                Array [
                  Array [
                    Object {
                      "justifyContent": "flex-end",
                    },
                    undefined,
                  ],
                  Object {
                    "backgroundColor": "#cccccc",
                    "borderRadius": 16,
                  },
                ],
              ]
            }
            thumbTintColor="#999999"
            tintColor="#cccccc"
            value={false}
          />
        </View>
      </View>
      <View
        accessibilityLabel="Clear sensor history button, Double tap to activate"
        accessible={true}
        focusable={true}
        importantForAccessibility="yes"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "#000000",
            "borderRadius": 44,
            "elevation": 2,
            "flexDirection": "row",
            "justifyContent": "center",
            "marginTop": 41.6655,
            "maxWidth": 675,
            "minWidth": 375,
            "opacity": 1,
            "paddingHorizontal": 15,
            "paddingVertical": 22,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.3,
            "shadowRadius": 2,
            "width": 600,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            importantForAccessibility="yes"
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#ffffff",
                  "fontFamily": "SuisseIntl",
                  "fontSize": 26,
                  "textAlign": "center",
                  "textAlignVertical": "center",
                },
                undefined,
              ]
            }
          >
            Clear sensor history
          </Text>
        </View>
      </View>
      <View
        accessibilityLabel="Reset Max/Min values button, Double tap to activate"
        accessible={true}
        focusable={true}
        importantForAccessibility="yes"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "#000000",
            "borderRadius": 44,
            "elevation": 2,
            "flexDirection": "row",
            "justifyContent": "center",
            "marginTop": 41.6655,
            "maxWidth": 675,
            "minWidth": 375,
            "opacity": 1,
            "paddingHorizontal": 15,
            "paddingVertical": 22,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.3,
            "shadowRadius": 2,
            "width": 600,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            importantForAccessibility="yes"
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#ffffff",
                  "fontFamily": "SuisseIntl",
                  "fontSize": 26,
                  "textAlign": "center",
                  "textAlignVertical": "center",
                },
                undefined,
              ]
            }
          >
            Reset max/min values
          </Text>
        </View>
      </View>
      <View
        accessibilityLabel="Clear history cache button, Double tap to activate"
        accessible={true}
        focusable={true}
        importantForAccessibility="yes"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "#000000",
            "borderRadius": 44,
            "elevation": 2,
            "flexDirection": "row",
            "justifyContent": "center",
            "marginTop": 41.6655,
            "maxWidth": 675,
            "minWidth": 375,
            "opacity": 1,
            "paddingHorizontal": 15,
            "paddingVertical": 22,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.3,
            "shadowRadius": 2,
            "width": 600,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            importantForAccessibility="yes"
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#ffffff",
                  "fontFamily": "SuisseIntl",
                  "fontSize": 26,
                  "textAlign": "center",
                  "textAlignVertical": "center",
                },
                undefined,
              ]
            }
          >
            Clear history cache
          </Text>
        </View>
      </View>
      <View
        accessibilityLabel="Mark node as failed button, Double tap to activate"
        accessible={true}
        focusable={true}
        importantForAccessibility="yes"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "#b5b5b5",
            "borderRadius": 44,
            "elevation": 2,
            "flexDirection": "row",
            "justifyContent": "center",
            "marginTop": 41.6655,
            "maxWidth": 675,
            "minWidth": 375,
            "opacity": 1,
            "paddingHorizontal": 15,
            "paddingVertical": 22,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.5,
            "shadowRadius": 2,
            "width": 600,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            importantForAccessibility="yes"
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#f5f5f5",
                  "fontFamily": "SuisseIntl",
                  "fontSize": 26,
                  "textAlign": "center",
                  "textAlignVertical": "center",
                },
                undefined,
              ]
            }
          >
            Mark node as failed
          </Text>
        </View>
      </View>
      <View
        accessibilityLabel="Exclude device button, Double tap to activate"
        accessible={true}
        focusable={true}
        importantForAccessibility="yes"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "#b5b5b5",
            "borderRadius": 44,
            "elevation": 2,
            "flexDirection": "row",
            "justifyContent": "center",
            "marginTop": 41.6655,
            "maxWidth": 675,
            "minWidth": 375,
            "opacity": 1,
            "paddingHorizontal": 15,
            "paddingVertical": 22,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.5,
            "shadowRadius": 2,
            "width": 600,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            importantForAccessibility="yes"
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#f5f5f5",
                  "fontFamily": "SuisseIntl",
                  "fontSize": 26,
                  "textAlign": "center",
                  "textAlignVertical": "center",
                },
                undefined,
              ]
            }
          >
            Exclude device
          </Text>
        </View>
      </View>
      <Text
        allowFontScaling={false}
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#555555",
            "fontFamily": "SuisseIntl",
            "fontSize": 30,
            "marginHorizontal": 41.6655,
            "marginTop": 41.6655,
            "textAlign": "center",
          }
        }
      >
        History cache is stored on your device to shorten load times. Clear the cache if you want to reload the values from the database
        .
      </Text>
      <Text
        allowFontScaling={false}
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#777777",
            "fontFamily": "SuisseIntl",
            "fontSize": 30,
            "marginTop": 41.6655,
          }
        }
      >
        Technical information
      </Text>
      <View
        accessibilityLabel="Protocol, Z-Wave , "
        accessible={true}
        importantForAccessibility="yes"
        style={
          Array [
            Object {
              "backgroundColor": "#f5f5f5",
              "borderRadius": 2,
              "elevation": 2,
              "marginTop": 10.416375,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 1,
                "width": 0,
              },
              "shadowOpacity": 0.23,
              "shadowRadius": 2,
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderRadius": 2,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "overflow": "hidden",
                "padding": 30,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "flexDirection": "row",
                "justifyContent": "flex-start",
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#000",
                    "fontFamily": "SuisseIntl",
                    "fontSize": 30,
                    "justifyContent": "center",
                  },
                  undefined,
                ]
              }
            >
              Protocol
            </Text>
          </View>
          <View
            accessible={true}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "flex-end",
                "opacity": 1,
                "paddingVertical": 8,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#4C4C4C",
                    "flex": 1,
                    "fontFamily": "SuisseIntl",
                    "fontSize": 30,
                    "textAlign": "right",
                    "textAlignVertical": "center",
                  },
                  undefined,
                ]
              }
            >
              Z-Wave
               
            </Text>
          </View>
        </View>
      </View>
      <View
        accessibilityLabel="Model, n/a , "
        accessible={true}
        importantForAccessibility="yes"
        style={
          Array [
            Object {
              "backgroundColor": "#f5f5f5",
              "borderRadius": 2,
              "elevation": 2,
              "marginTop": 10.416375,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 1,
                "width": 0,
              },
              "shadowOpacity": 0.23,
              "shadowRadius": 2,
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderRadius": 2,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "overflow": "hidden",
                "padding": 30,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "flexDirection": "row",
                "justifyContent": "flex-start",
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#000",
                    "fontFamily": "SuisseIntl",
                    "fontSize": 30,
                    "justifyContent": "center",
                  },
                  undefined,
                ]
              }
            >
              Model
            </Text>
          </View>
          <View
            accessible={true}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "flex-end",
                "opacity": 1,
                "paddingVertical": 8,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#4C4C4C",
                    "flex": 1,
                    "fontFamily": "SuisseIntl",
                    "fontSize": 30,
                    "textAlign": "right",
                    "textAlignVertical": "center",
                  },
                  undefined,
                ]
              }
            >
              n/a
               
            </Text>
          </View>
        </View>
      </View>
      <View
        accessibilityLabel="Sensor ID, 69 , "
        accessible={true}
        importantForAccessibility="yes"
        style={
          Array [
            Object {
              "backgroundColor": "#f5f5f5",
              "borderRadius": 2,
              "elevation": 2,
              "marginTop": 10.416375,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 1,
                "width": 0,
              },
              "shadowOpacity": 0.23,
              "shadowRadius": 2,
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderRadius": 2,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "overflow": "hidden",
                "padding": 30,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "flexDirection": "row",
                "justifyContent": "flex-start",
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#000",
                    "fontFamily": "SuisseIntl",
                    "fontSize": 30,
                    "justifyContent": "center",
                  },
                  undefined,
                ]
              }
            >
              Sensor ID
            </Text>
          </View>
          <View
            accessible={true}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "flex-end",
                "opacity": 1,
                "paddingVertical": 8,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#4C4C4C",
                    "flex": 1,
                    "fontFamily": "SuisseIntl",
                    "fontSize": 30,
                    "textAlign": "right",
                    "textAlignVertical": "center",
                  },
                  undefined,
                ]
              }
            >
              69
               
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
